packages/cli/src/modules/insights/__tests__/insights-pruning.service.test.ts

```diff
@@ -0,0 +1,169 @@
+import { Container } from '@n8n/di';
+import { mock } from 'jest-mock-extended';
+import { DateTime } from 'luxon';
+
+import { Time } from '@/constants';
+import { mockLogger } from '@test/mocking';
+import { createTeamProject } from '@test-integration/db/projects';
+import { createWorkflow } from '@test-integration/db/workflows';
+import * as testDb from '@test-integration/test-db';
+
+import {
+	createCompactedInsightsEvent,
+	createMetadata,
+} from '../database/entities/__tests__/db-utils';
+import { InsightsByPeriodRepository } from '../database/repositories/insights-by-period.repository';
+import { InsightsPruningService } from '../insights-pruning.service';
+import { InsightsConfig } from '../insights.config';
+
+// Initialize DB once for all tests
+beforeAll(async () => {
+	await testDb.init();
+});
+
+describe('InsightsPrunningService', () => {
+	let insightsConfig: InsightsConfig;
+	let insightsByPeriodRepository: InsightsByPeriodRepository;
+	let insightsPruningService: InsightsPruningService;
+	beforeAll(async () => {
+		insightsConfig = Container.get(InsightsConfig);
+		insightsConfig.maxAgeDays = 10;
+		insightsConfig.pruneCheckIntervalHours = 1;
+		insightsPruningService = Container.get(InsightsPruningService);
+		insightsByPeriodRepository = Container.get(InsightsByPeriodRepository);
+	});
+
+	test('old insights get pruned successfully', async () => {
+		// ARRANGE
+		const project = await createTeamProject();
+		const workflow = await createWorkflow({}, project);
+
+		await createMetadata(workflow);
+
+		const timestamp = DateTime.utc().minus({ days: insightsConfig.maxAgeDays + 1 });
+		await createCompactedInsightsEvent(workflow, {
+			type: 'success',
+			value: 1,
+			periodUnit: 'day',
+			periodStart: timestamp,
+		});
+
+		// ACT
+		await insightsPruningService.pruneInsights();
+
+		// ASSERT
+		expect(await insightsByPeriodRepository.count()).toBe(0);
```

Comment:
I wonder if there's a difference between this and `resolves`.

Reply:
Apparently, .resolves handle more fgracefully the failure case. I'll change to it